core/src/main/java/com/scalar/db/transaction/consensuscommit/RecoveryHandler.java

```diff
@@ -84,31 +81,13 @@ public void recover(Selection selection, TransactionResult result) {
   Optional<TransactionResult> getLatestResult(Selection selection, TransactionResult result)
       throws ExecutionException {
     Get get =
-        new Get(
-                getPartitionKey(selection, result),
-                getClusteringKey(selection, result).orElse(null))
+        new Get(result.getPartitionKey().get(), result.getClusteringKey().orElse(null))
```

Comment:
We should use the partition key and the clustering key from the `result` because the partition key from the `selection` can be index key, and in this case, the Get operation can return multiple results (and, in that case, the Get operation fails).